cfme/tests/networks/test_provision_to_virtual_network.py

```diff
@@ -0,0 +1,74 @@
+from widgetastic.utils import partial_match
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.common.vm import VM
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.provisioning import do_vm_provisioning
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.provider([RHEVMProvider],
+                         required_fields=[['provisioning', 'template']],
+                         scope="module"
+                         ),
+    pytest.mark.uncollectif(lambda provider: provider.version < 4.2,
+                            reason="ovn functionality is limited in 4.1"),
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.ignore_stream("5.8"),
+    test_requirements.provision,
+]
+
+
+@pytest.fixture(scope='function')
+def network(request, provider, appliance):
+    """Test adding cloud network in ui."""
+    request.addfinalizer(
+        lambda: network.delete()
+    )
+
+    test_name = "test_network_{}".format(fauxfactory.gen_alphanumeric(6))
+    net_manager = "{} Network Manager".format(provider.name)
+    collection = appliance.collections.network_providers
+    network_provider = collection.instantiate(
+        name=net_manager
+    )
+    collection = appliance.collections.cloud_networks
+    network = collection.create(test_name, "tenant", network_provider, net_manager, "None")
+    return network
+
+
+def test_provision_vm_to_virtual_network(appliance, setup_provider, provider,
+                                 vm_name, request, provisioning, network):
+    """ Tests provisioning a vm from a template to a virtual network
+
+    Metadata:
+        test_flag: provision
+    """
+
+    request.addfinalizer(
+        lambda: VM.factory(vm_name, provider).cleanup_on_provider())
+
+    template = provisioning['template']
+    provisioning_data = {
+        'catalog': {
+            'vm_name': vm_name
+        },
+        'environment': {
+            'vm_name': vm_name,
+            'automatic_placement': True
+        },
+        'network': {
+            'vlan': partial_match(network.name)
+        }
+    }
+    wait_for(
```

Comment:
Required: Add message in case if wait_for is failed
 wait_for(do_vm_provisioning,
               message='Cannot do provision for vm {}'.format(vm_ma,e)

Reply:
Done.